app/views/compute_resources_vms/form/openstack/_base.html.erb

```diff
@@ -18,3 +18,10 @@
 <%= selectable_f f, :network, compute_resource.address_pools, { include_blank: 'None' }, { :label => _("Floating IP network"), :label_size => "col-md-2" } %>
 <%= checkbox_f f, :boot_from_volume, {:label => _("Boot from volume"), :label_size => "col-md-2", :help_inline => _("Create new boot volume from image")}, "true", "false" %>
 <%= text_f f, :size_gb, { :label => _("New boot volume size (GB)"), :label_size => "col-md-2", :help_inline => _("Defaults to image size if left blank") } %>
+<%= selectable_f f, :scheduler_hint_filter, compute_resource.possible_scheduler_hints, { include_blank: 'None' }, { :label => _("Scheduler hint filter"), :label_size => "col-md-2", :'data-url' => scheduler_hint_selected_hosts_path,
+                                                                                                              :onchange => 'schedulerHintFilterSelected(this);' } %>
+<div id="scheduler_hint_wrapper">
+  <%= if @host && @host.compute_attributes[:scheduler_hint_filter]
```

Comment:
I think this `if` should wrap the `div` too, otherwise you'll create it for no reason.

Secondly, `@host` is always going to be available here isn't it? Can't think right now of a situation where `@host` would not be present, if you can think of one please comment otherwise remove it from the condition.

Lastly, can the `scheduler_hint_filters` be changed when you edit an already created host? If not, make sure to disable it, a common way to do this is to pass a local variable to `render` called `new_host`, which you can generate with `new_host?(host)`